crates/astria-sequencer-utils/src/genesis_parser.rs

```diff
@@ -0,0 +1,68 @@
+use std::{
+    fs::File,
+    path::Path,
+};
+
+use anyhow::{
+    Context as _,
+    Result,
+};
+use serde_json::{
+    to_writer_pretty,
+    Value,
+};
+use tracing::{
+    info,
+    instrument,
+};
+
+use crate::config::GenesisParserArgs;
+
+pub struct GenesisParser;
+
+impl GenesisParser {
+    #[instrument(skip_all)]
+    pub async fn propigate_data(data: GenesisParserArgs) -> Result<()> {
+        info!(
+            source_genesis_file = data.source_genesis_file.as_str(),
+            destination_genesis_file = data.destination_genesis_file.as_str(),
+            "loading genesis json data for propigation"
+        );
+        // load sequencer genesis data
+        let source_genesis_file_path = File::open(data.source_genesis_file)
+            .context("failed to open sequencer genesis file")?;
+        let source_genesis_data: Value = serde_json::from_reader(&source_genesis_file_path)
+            .context("failed deserializing sequencer genesis state from file")?;
+
+        // load cometbft genesis data
+        let destination_genesis_file_path = File::open(data.destination_genesis_file.clone())
```

Comment:
```suggestion
        let destination_genesis_file_path = File::open(&data.destination_genesis_file)
```
nbd for a small binary but the clone isn't needed

Reply:
removed the clone and replaced with .as_str(). the dest path is used in a few places